js/reference.js

```diff
@@ -322,6 +347,47 @@ var ERMrest = (function(module) {
         },
 
         /**
+         * This is a private funtion that checks the user permissions for modifying the affiliated entity, record or table
+         * Sets a property on the reference object used by canCreate/canUpdate/canDelete
+         * @memberof ERMrest
+         * @private
+         */
+         _checkPermissions: function (permission) {
+            var editCatalog = false,
+                ignoreRecord = false,
+                ignoreUri = module._annotations.IGNORE,
+                metadata = this._meta[permission],
```

Comment:
It's confusing to call this variable `metadata`. It would be more accurate to call it `acl`, for access control list.